src/event/event_set.rs

```diff
@@ -131,24 +70,48 @@ impl EventSet {
     /// `()` if successful, `RuntimeError` otherwise.
     #[allow(dead_code)]
     pub(crate) fn deactivate_event(&self, event_id: EventId) -> Result<(), RuntimeError> {
-        self.event_states.lock(|event_states| {
-            match event_states
-                .iter_mut()
-                .find(|event_state| event_state.id() == event_id)
-            {
-                Some(event_state) => event_state.set_active(false),
-                None => return Err(RuntimeError::EventNotFound(event_id)),
-            };
-
-            Ok(())
+        self.event_queue.lock(|event_queue| {
```

Comment:
that whole lock looks like a giant https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.filter

you have iterators here, so you should be able to use it to one-line this whole function

Reply:
closing after consultation; i forgot we don't have heap here